contracts/protocol/pool/PoolAAPositionMove.sol

```diff
@@ -0,0 +1,242 @@
+                    _calculateLiquidityAndDebtIndex(reserveCache);
+                    vars.xTokenAddresses[j] = reserveCache.xTokenAddress;
+                    vars.assetTypes[j] = reserveCache
+                        .reserveConfiguration
+                        .getAssetType();
+                    vars.liquidityIndex[j] = reserveCache.nextLiquidityIndex;
+                    vars.debtIndex[j] = reserveCache.nextVariableBorrowIndex;
+                }
+
+                if (vars.assetTypes[j] == DataTypes.AssetType.ERC20) {
+                    //handle ptoken
+                    {
+                        IPToken pToken = IPToken(vars.xTokenAddresses[j]);
+                        uint256 balance = pToken.balanceOf(user);
+                        if (balance > 0) {
+                            pToken.burn(
+                                user,
+                                vars.xTokenAddresses[j],
+                                balance,
+                                vars.liquidityIndex[j],
+                                timeLockParams
+                            );
+                            pToken.mint(
+                                aaAccount,
+                                aaAccount,
+                                balance,
+                                vars.liquidityIndex[j]
+                            );
+                            if (userConfig.isUsingAsCollateral(j)) {
+                                aaConfig.setUsingAsCollateral(j, true);
+                                userConfig.setUsingAsCollateral(j, false);
+                            }
+                        }
+                    }
+
+                    //handle debt token
+                    {
+                        IVariableDebtToken debtToken = IVariableDebtToken(
+                            vars.debtTokenAddresses[j]
+                        );
+                        uint256 balance = debtToken.balanceOf(user);
+                        if (balance > 0) {
+                            debtToken.burn(user, balance, vars.debtIndex[j]);
+                            debtToken.mint(
+                                aaAccount,
+                                aaAccount,
+                                balance,
+                                vars.debtIndex[j]
+                            );
+                            aaConfig.setBorrowing(j, true);
+                            userConfig.setBorrowing(j, false);
+                        }
+                    }
+                } else {
+                    INToken nToken = INToken(vars.xTokenAddresses[j]);
+                    uint256 balance = nToken.balanceOf(user);
+                    for (uint256 k = 0; k < balance; k++) {
+                        uint256 tokenId = nToken.tokenOfOwnerByIndex(user, k);
+                        bool isCollateral = nToken.isUsedAsCollateral(tokenId);
+                        nToken.transferOnLiquidation(user, aaAccount, tokenId);
```

Comment:
if user have staking NFT in current account , user may use positionMoveToAA to keep all staking amount of $Ape in current account , move others token to AA.